### PR TITLE
12 mig base optional quick create mgs

### PR DESCRIPTION
**Pull request by GillesTephaneMeyomesse**
https://github.com/OCA/server-ux/issues/24

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash together translation commits (and maybe some meaningless of the same author)

**Comment by pedrobaeza** [comment]
We have been talking about deprecating this module in favor of web_m2x_options expanded with definition per model.

cc @sbidoul

**Review by i-vyshnevska — APPROVED** [review]
LG, possible solution  about  web_m2x_options could be mentioned in Readme and added to Roadmap

**Review by simahawk — CHANGES_REQUESTED** [review]
LG but I'd squash a lot of useless commits like "descr" or similar together w/ the main ones.

Regarding tackling this on widget side: I agree.
I think this should be improved and instead of bumping an error in user's face it should block creation by setting `no_create` option on the views (`fields_view_get` or m2o widget override).

That being said we could track this in the roadmap and plan it for a next dev.

**Comment by fmdl** [comment]
Note : I made this PR https://github.com/odoo/odoo/pull/32013 for Odoo to make this module for simpler.

**Comment by alexis-via** [comment]
I proposed an improvement to this PR: https://github.com/acsone/server-ux/pull/1

**Comment by alexis-via** [comment]
FYI, I will stop using base_optional_quick_create and use web_m2x_options instead. I updated my module eradicate_quick_create to depend on web_m2x_options instead of base_optional_quick_create https://github.com/akretion/odoo-usability/blob/12.0/eradicate_quick_create/__manifest__.py

**Review by SimoRubi — APPROVED** [review]
Tested locally and it works, thanks!
Could you please squash a bit the commits following https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests?

File: base_optional_quick_create/models/ir_model.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    model_obj._revert_method(method_name)
+        return True
+
+    def _register_hook(self):
+        models = self.search([])
+        models._patch_quick_create()
+        return super()._register_hook()
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
```
Comment (fmdl) [inline base_optional_quick_create/models/ir_model.py:44]: should be remove

File: base_optional_quick_create/models/ir_model.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                if method and hasattr(method, 'origin'):
+                    model_obj._revert_method(method_name)
+        return True
+
+    def _register_hook(self):
+        models = self.search([])
+        models._patch_quick_create()
+        return super()._register_hook()
+
+    @api.model
```
Comment (fmdl) [inline base_optional_quick_create/models/ir_model.py:43]: use 

```python
    @api.model_create_multi
    def create(self, vals_list):
         ir_models = super().create(vals_list)
         ir_models._patch_quick_create()
         return ir_models
```

**Comment by ougc27** [comment]
Hello  @GillesTephaneMeyomesse this PR will be merged soon?

I can help you with the last comments, what do you think?

**Comment by StefanRijnhart** [comment]
> I think this should be improved and instead of bumping an error in user's face

@simahawk this module does not throw an error in the user's face. Instead, it uses the mechanism in the Odoo web client that triggers the fallback to create_and_edit when whatever error occurs during quick create: https://github.com/odoo/odoo/blob/12.0/addons/web/static/src/js/fields/relational_fields.js#L391-L398

I don't see any progress on the integration of this per-module setting in web_m2x_options and I don't really see how that should work so I'm going to propose an improved version of this PR later today.

**Comment by StefanRijnhart** [comment]
Continuing here: https://github.com/OCA/server-ux/pull/160. @fmdl I fixed your comments, thanks!

**Pull request closed without merging**